File: android/app/src/main/java/io/github/motelume/ddlcard/core/CountdownFormatter.kt
package io.github.motelume.ddlcard.core

import kotlin.math.abs

object CountdownFormatter {
    fun format(dueAtUtc: Long, nowUtc: Long = System.currentTimeMillis()): String {
        val overdue = dueAtUtc < nowUtc
        val totalMinutes = abs(dueAtUtc - nowUtc) / 60_000
        val days = totalMinutes / 1_440
        val hours = totalMinutes % 1_440 / 60
        val minutes = totalMinutes % 60
        val value = if (days > 0) "${days}天 ${hours}小时 ${minutes}分钟" else "${hours}小时 ${minutes}分钟"
        return if (overdue) "已逾期 $value" else "还剩 $value"
    }
}

File: src/DDLCard.Core/Services/CountdownFormatter.cs
namespace DDLCard.Core.Services;

public static class CountdownFormatter
{
    public static string Format(DateTimeOffset dueAtUtc, DateTimeOffset nowUtc)
    {
        var overdue = dueAtUtc < nowUtc;
        var duration = (overdue ? nowUtc - dueAtUtc : dueAtUtc - nowUtc).Duration();
        var totalMinutes = Math.Max(0, (int)Math.Floor(duration.TotalMinutes));
        var days = totalMinutes / 1440;
        var hours = totalMinutes % 1440 / 60;
        var minutes = totalMinutes % 60;
        var value = days > 0
            ? $"{days}天 {hours}小时 {minutes}分钟"
            : $"{hours}小时 {minutes}分钟";

        return overdue ? $"已逾期 {value}" : $"还剩 {value}";
    }
}

